### ISSUE TITLE
sometimes loss is negative during training model

### ISSUE THREAD

**Issue by hw118118** [caogang/wgan-gp#45]
I use Tensorflow to do the work.And my loss:
self.g_loss = -tf.reduce_mean(d_logits_fake)
self.d_loss = tf.reduce_mean(d_logits_fake) - tf.reduce_mean(d_logits_real)+GP
but during traning,I find sometimes g_loss is negative.I don't know why i will get it.Who will explain it

**Comment by silentsaber** [comment]
I think that's when training generator, we want it produce fake data to fool the discriminator, so the d_logits_fake will be quite large like the real data. So d_logits_fake will be a positive val.
when training discriminator, d_logits_fake is quite small instead. That's mean it is  a negtive value
